### PR TITLE
Unminting

### PR DESCRIPTION
**Pull request by r-czajkowski**
This PR adds the Unmint page where users can request redemption. Here we only implement UI this PR doesn't cover contract integration- we are going to address it in a separate PR.

### Main changes
- create Unmint page- display a unmint form where users can request a redemption,
- create `InitiateUnminting` modal- confirmation modal where users can validate the redemption details before signing the transaction,
- extract the `Step` component (previously `TimelineItem` component from `src/pages/tBTC/Bridge/components/TimelineItem.tsx`) to the shared `components` dir. Implemented using the compound component pattern so we can keep the same style but we have more flexibility in terms of positioning some parts of this component (eg. 2 badges instead of only one). It's a component from the [Threshold Design System](https://www.figma.com/file/zZi2fYDUjWEMPQJWAt8VWv/Threshold-DS?type=design&node-id=874-18470&t=5ynOWzbkUwXIn9AT-0),
- extract `TransactionDetails` components to a shared `components` dir,
- rename some components from `Minting...` to `BridgeProcess...` so we can use them in both contexts- `mint` and `unmint`.

<details>
  <summary>Screenshots</summary>
  
obraz

obraz

</details>

### CODE REVIEW

File: src/pages/tBTC/Bridge/Unmint.tsx [hunk 1]
```diff
@@ -1,10 +1,220 @@
...
+    "unmint",
+    bitcoinNetwork
+  )
+  return (
+    <Form mt={10}>
+      <FormikTokenBalanceInput
+        name="amount"
+        label={
+          // TODO: Extract to a shared component- the same layout is used in
+          // `TokenAmountForm` and `UnstakingFormLabel` components.
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/Unmint.tsx:150]: Let's not forget about this TODO

File: src/pages/tBTC/Bridge/Unmint.tsx [hunk 2]
```diff
@@ -1,10 +1,220 @@
...
+          </>
+        }
+        placeholder="Amount you want to unmint"
+        max={maxTokenAmount}
+        icon={tBTCFillBlack}
+      />
+      <FormikInput
+        name="btcAddress"
+        label="BTC Address"
+        // TODO: add BTC addresse prefixes
```
Comment (michalsmiarowski) [inline src/pages/tBTC/Bridge/Unmint.tsx:170]: I think this TODO can be removed

File: src/components/Modal/tBTC/InitiateUnminting.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}) => {
+  // TODO: calculate the BTC amount- take into account fees
+  const btcAmount = "1.25"
+  const thresholdNetworkFee = "0"
+  const btcMinerFee = "0"
+
+  // TODO: implement submit function
+  const initiateUnminting = () => {
+    console.log("unminting")
+  }
```
Comment (michalsmiarowski) [inline src/components/Modal/tBTC/InitiateUnminting.tsx:42]: I assume those two will be implemented in a separate PR, but let's not forget about those two TODOs

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Looking great 🔥 Left some comments to look at before the merge

File: src/pages/tBTC/Bridge/Unmint.tsx [hunk 4]
```diff
@@ -1,10 +1,220 @@
...
+    })
+  }
+
+  return (
+    <BridgeLayout>
+      <BridgeLayoutMainSection>
+        <BridgeProcessCardTitle bridgeProcess="unmint" />
+        <BridgeProcessCardSubTitle
+          stepText="Step 1"
+          subTitle="Unmint your tBTC tokens"
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/Unmint.tsx:76]: Yeah. We can probably use the same empty state as for the `Mint` section but I have to ask designers. I think we can address it in a separate PR.

File: src/pages/tBTC/Bridge/Unmint.tsx [hunk 5]
```diff
@@ -1,10 +1,220 @@
...
+import { useModal } from "../../../hooks/useModal"
 
 export const UnmintPage: PageComponent = ({}) => {
-  return <UnmintingCard gridArea="main-card" p={35} />
+  const { balance } = useToken(Token.TBTCV2)
+  const { openModal } = useModal()
+  const threshold = useThreshold()
+
+  const onSubmitForm = (values: UnmintFormValues) => {
+    console.log("on submittttt")
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/Unmint.tsx:63]: I'll implement the submit functions in a separate PR- I'm currently working on the contract integration.

File: src/utils/tBTC.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    mainnet: MINTING_MAINNET_BTC_RECOVERY_ADDRESS_PREFIXES,
+    testnet: MINTING_TESTNET_BTC_RECOVERY_ADDRESS_PREFIXES,
+  },
+  unmint: {
+    mainnet: UNMINTING_MAINNET_BTC_ADDRESS_PREFIXES,
+    testnet: UNMINTING_TESTNET_BTC_ADDRESS_PREFIXES,
+  },
+} as const
+
+const getSupportedAddressPrefixes = (
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/utils/tBTC.ts]: 84761d1

File: src/pages/tBTC/Bridge/Unmint.tsx (hunk 2 above)
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/Unmint.tsx]: ef78736

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🔥

**Pull request merged**